### PR TITLE
bf: sanity check on returned content-length

### PR DESCRIPTION
**Pull request by jonathan-gramain**
In responseStreamData() helper, ensure the content-length is correct
with respect to data locations aggregated size. Log an error and
return internal error to the client if not.

This should catch off-by-one errors when computing ranges of data
locations to fetch and return.

### CODE REVIEW

**Comment by jonathan-gramain** [comment]
E2E passed in http://ci.ironmann.io/gh/scality/Integration/16632

File: lib/s3routes/routesUtils.js [hunk 1]
```diff
@@ -402,6 +411,21 @@ const routesUtils = {
...
                                                     resHeaders);
         }
+        if (dataLocations !== null && !response.headersSent) {
+            // sanity check of content length against individual data
+            // locations to fetch
+            const contentLength = resHeaders && resHeaders['Content-Length'];
+            if (contentLength !== undefined &&
+                !_contentLengthMatchesLocations(contentLength,
+                                                dataLocations)) {
+                log.error('logic error: total length of fetched data ' +
```
Comment (rahulreddy) [inline lib/s3routes/routesUtils.js]: May be remove `logic error:` and keep the rest of the message?

Comment (jonathan-gramain) [reply lib/s3routes/routesUtils.js]: ok

Comment (jonathan-gramain) [reply lib/s3routes/routesUtils.js]: Well, maybe not going to change it since master branch is already merged.

**Comment by rahulreddy** [comment]
@ironman-machine try

**Comment by rahulreddy** [comment]
@ironman-machine r+

**Pull request merged**